### PR TITLE
Label test structs with "wanted"

### PR DESCRIPTION
**Pull request by richsalz**
So that the test failure messages are a little more obvious.

Based on discussion from #3506.

### CODE REVIEW

File: test/evp_test.c [hunk 1]
```diff
@@ -66,9 +66,9 @@ struct evp_test_method_st {
  * Linked list of named keys.
  */
 typedef struct key_list_st {
+    struct key_list_st *next;
     char *name;
     EVP_PKEY *key;
-    struct key_list_st *next;
```
Comment (levitte) [inline test/evp_test.c]: Errr, because why?

Comment (richsalz) [inline test/evp_test.c]: I like the overhead out of the way first.  Shrug.  Put it back to the last element.  Would like @dot-asm to review before I merge, since this is all his doing :)

Comment (levitte) [inline test/evp_test.c]: Ok...  it's just a bit confusing to see these unrelated changes being "sneaked in"

File: test/evp_test.c [hunk 2]
```diff
@@ -689,7 +681,7 @@ static int cipher_test_enc(EVP_TEST *t, int enc,
         t->err = "CIPHERFINAL_ERROR";
         goto err;
     }
-    if (!compare_mem(out, out_len, tmp + out_misalign, tmplen + tmpflen)) {
+    if (!TEST_mem_eq(out, out_len, tmp + out_misalign, tmplen + tmpflen)) {
```
Comment (dot-asm) [inline test/evp_test.c]: Why note rename even `out`?

File: test/evp_test.c [hunk 3]
```diff
@@ -705,7 +697,7 @@ static int cipher_test_enc(EVP_TEST *t, int enc,
             t->err = "TAG_RETRIEVE_ERROR";
             goto err;
         }
-        if (!compare_mem(cdat->tag, cdat->tag_len, rtag, cdat->tag_len)) {
+        if (!TEST_mem_eq(cdat->tag, cdat->tag_len, rtag, cdat->tag_len)) {
```
Comment (dot-asm) [inline test/evp_test.c]: Why not rename `cdat-tag` and `rtag`?

File: test/evp_test.c (hunk 2 above)
Comment (richsalz) [inline test/evp_test.c]: done.

File: test/evp_test.c (hunk 3 above)
Comment (richsalz) [inline test/evp_test.c]: done.

**Comment by richsalz** [comment]
I also renamed most of the "[cpfk]data" variables to test.  You now see failures like this:
```
    # --- wanted
    # +++ tmp + out_misalign
    # 0000:-3cf55d656e9c0664 513359
    # 0000:+3cf55d656e9c0664 513358
    #                            ^^
 
   # ERROR: (memory) 'test->tag == received_tag' failed @ test/evp_test.c:701
    # --- test->tag
    # +++ received_tag
    # 0000:-f0723bce23528ce6 ccb10ff9627038bf
    # 0000:+e0723bce23528ce6 ccb10ff9627038bf
    #       ^^

**Comment by dot-asm** [comment]
And perhaps push???

**Comment by richsalz** [comment]
```
    # ERROR: (memory) 'test->wanted == md' failed @ test/evp_test.c:395
    # --- test->wanted
    # +++ md
    # 0000:-29fa61d75522a466 9b44e39c1d2e1726 c530232130d407f8 9afee0964997f7a7
    # 0000:+19fa61d75522a466 9b44e39c1d2e1726 c530232130d407f8 9afee0964997f7a7
    #       ^^
    # 0020: 3e83be698b288feb cf88e3e03c4f0757 ea8964e59b63d937 08b138cc42a66eb3

    # INFO:  @ test/evp_test.c:2146
    # Above error from the test at test/evpdigest.txt:266 unexpected error DIGEST_MISMATCH
    # ERROR: (memory) 'test->input == decode_out' failed @ test/evp_test.c:1587
    # --- test->input
    # +++ decode_out
    # 0000:-4f70656e53534c4f 70656e53534c0a
    # 0000:+6b70656e53534c4f 70656e53534c0a
    #       ^^

    # INFO:  @ test/evp_test.c:2146
    # Above error from the test at test/evpencod.txt:190 unexpected error BAD_DECODING

```

**Comment by richsalz** [comment]
Additional commit (to be squashed if I remember) pushed.

File: test/evp_test.c [hunk 4]
```diff
@@ -781,9 +774,9 @@ static int cipher_test_run(EVP_TEST *t)
              * XTS, CCM and Wrap modes have special requirements about input
              * lengths so we don't fragment for those
              */
-            if (cdat->aead == EVP_CIPH_CCM_MODE
-                    || EVP_CIPHER_mode(cdat->cipher) == EVP_CIPH_XTS_MODE
-                     || EVP_CIPHER_mode(cdat->cipher) == EVP_CIPH_WRAP_MODE)
+            if (test->aead == EVP_CIPH_CCM_MODE
+                    || EVP_CIPHER_mode(test->cipher) == EVP_CIPH_XTS_MODE
+                     || EVP_CIPHER_mode(test->cipher) == EVP_CIPH_WRAP_MODE)
```
Comment (dot-asm) [inline test/evp_test.c]: Why not fix jagged indentation?

**Comment by dot-asm** [comment]
> ERROR: (memory) 'test->tag == received_tag'

I don't see that `test->something` is better. But `wanted->something` would be... Which means that instead of renaming fields one can rename variables...

**Comment by dot-asm** [comment]
Sorry about closing/re-opening. Wrong button.

File: test/evp_test.c [hunk 5]
```diff
@@ -689,23 +681,24 @@ static int cipher_test_enc(EVP_TEST *t, int enc,
         t->err = "CIPHERFINAL_ERROR";
         goto err;
     }
-    if (!compare_mem(out, out_len, tmp + out_misalign, tmplen + tmpflen)) {
+    if (!TEST_mem_eq(wanted, wanted_len,
+                     tmp + out_misalign, tmplen + tmpflen)) {
```
Comment (dot-asm) [inline test/evp_test.c]: Suggestion to rename `*dat` to `wanted` presents challenge here. One can use `_wanted` or `expected`...

File: test/evp_test.c (hunk 4 above)
Comment (richsalz) [inline test/evp_test.c]: because like Islamic artists, some imperfection must be present? :)
Fixed.

**Comment by richsalz** [comment]
I used output_wanted

File: test/evp_test.c (hunk 5 above)
Comment (richsalz) [inline test/evp_test.c]: I used output_wanted

**Comment by richsalz** [comment]
updated commit pushed that renames things.

**Comment by dot-asm** [comment]
Instead of pointing out imperfections in my view I've chosen to simply create alternative #3550 and let you point out imperfections. [I'm not fond of 'wanted' as it sounds too "wild-west".]

**Comment by richsalz** [comment]
i assume you also didn't like the "evp-most-wanted" branch name :)
